src/main/java/org/pac4j/vertx/VertxWebContext.java

```diff
@@ -199,7 +199,12 @@ public String getFullRequestURL() {
 
     @Override
     public void addResponseCookie(Cookie cookie) {
-        routingContext.addCookie(io.vertx.ext.web.Cookie.cookie(cookie.getName(), cookie.getValue()));
+        routingContext.addCookie(io.vertx.ext.web.Cookie.cookie(cookie.getName(), cookie.getValue())
+                .setHttpOnly(cookie.isHttpOnly())
+                .setSecure(cookie.isSecure())
+                .setDomain(cookie.getDomain())
```

Comment:
By default, the `domain` and `path` are `null`: will it be properly supported?

Reply:
hello, it will be properly supported, the netty Cookie (io.netty.handler.codec.http.cookie.ServerCookieEncoder.encode)

does this:

`if (cookie.path() != null) {
            CookieUtil.add(buf, "Path", cookie.path());
        }
        if (cookie.domain() != null) {
            CookieUtil.add(buf, "Domain", cookie.domain());
        }
        if (cookie.isSecure()) {
            CookieUtil.add(buf, "Secure");
        }
        if (cookie.isHttpOnly()) {
            CookieUtil.add(buf, "HTTPOnly");
        }
        return CookieUtil.stripTrailingSeparator(buf);`

Reply:
but i have another problem, the class CsrfTokenGeneratorAuthorizer does not handle cookie maxAge, hence we cannot set maxAge

Reply:
hello, there is something important, if the CsrfTokenGeneratorAuthorizer  `maxAge`
 (have just added in pac4j) is not set, the default one of org.pac4j.core.context.Cookie  which has the value of -1 will be used, this will cause the cookie to be created with an expired date (current date - 1 millisecond), hence the browser will not create the cookie,
can we set the default maxAge to -9223372036854775808L as that of netty Cookie for the  
 encode method of netty

`if (cookie.maxAge() != -9223372036854775808L) {
            CookieUtil.add(buf, "Max-Age", cookie.maxAge());
            Date expires = new Date(cookie.maxAge() * 1000L + System.currentTimeMillis());
            buf.append("Expires");
            buf.append('=');
            DateFormatter.append(expires, buf);
            buf.append(';');
            buf.append(' ');
        }`

 not to create the `Expires` this will cause the cookie to never expire (as is the case now).

thanks;

Reply:
This is not fully clear to me, but yes, the default values of the pac4j `Cookie` component must be updated per pac4j implementation. Would you mind updating the PR to highlight that?

Reply:
OK. Got it. Let's do that.